File: csharp/examples/Request.cs
using System;
using System.IdentityModel.Tokens.Jwt;
using System.Security.Cryptography;
using System.Text;
using Microsoft.IdentityModel.Tokens;
using RestSharp;
using dotenv;

namespace Examples
{
    class Request
    {
        private static string _accessKey;
        private static string _secretKey;
        private static string _apiUrl;

        public Request() {
            dotenv.net.DotEnv.Load();
            _accessKey = Environment.GetEnvironmentVariable("ACCESS_KEY");
            _secretKey = Environment.GetEnvironmentVariable("SECRET_KEY");
            _apiUrl = Environment.GetEnvironmentVariable("API_URL");
        }

        public void GetRequest(string path, string queryParams)
        {
            var client = new RestClient(_apiUrl);
            var request = new RestRequest(path, Method.Get);
            request.RequestFormat = DataFormat.Json;
            GenerateJwtToken(ref request, path);
            var response = client.Get(request);
            Console.WriteLine(response.Content);
        }

        public void PostRequest(string path, string body)
        {
            var client = new RestClient(_apiUrl);
            var request = new RestRequest(path, Method.Post);
            request.RequestFormat = DataFormat.Json;
            GenerateJwtToken(ref request, path, body);
            var response = client.Post(request);
            Console.WriteLine(response.Content);
        }

        private void GenerateJwtToken(ref RestRequest request, string path, string body="")
        {
            var key         = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_secretKey));
            var credentials = new SigningCredentials(key, SecurityAlgorithms.HmacSha256);
            var header      = new JwtHeader(credentials)
            {
                { "access_key", _accessKey }
            };
            var payload = new JwtPayload
            {
                {"path", "/" + path},
                {"content", Sha256HexDigest(body) }
            };
            
            var securityToken = new JwtSecurityToken(header, payload);
            var handler       = new JwtSecurityTokenHandler();
            var token   = handler.WriteToken(securityToken);
            
            request.AddHeader("Authorization", "Bearer " + token);
            request.AddHeader("Content-Type", "application/json");
            if (body != "")
            {
                request.AddJsonBody(body);
            }
        }

        private static string Sha256HexDigest(string body)
        {
            return BitConverter.ToString(SHA256.Create().ComputeHash(Encoding.UTF8.GetBytes(body)))
                .Replace("-", String.Empty).ToLower();
        }
    }
}

File: javascript-axios/request.js
require('dotenv').config({ path: '../.env' });
const axios = require('axios');
const CryptoJS = require("crypto-js");
const sign = require('jwt-encode');


const API_URL		= process.env.API_URL;
const ACCESS_KEY 	= process.env.ACCESS_KEY;
const SECRET_KEY 	= process.env.SECRET_KEY;

exports.request = {
	token: function({ path, body = null }) {
		let token = "";
		let header = { access_key: ACCESS_KEY};
		let payload = {
			path: path
		};
		
		if (body) {
			payload.content = CryptoJS.SHA256(JSON.stringify(body)).toString(CryptoJS.enc.Hex);
		} else {
			payload.content = CryptoJS.SHA256(encodeURIComponent('')).toString(CryptoJS.enc.Hex);
		}
		token = sign(payload, SECRET_KEY, header);
		console.log(`token=${token}`);
		return token;
	},
	get: function ({ path, token, queryParams = '' }) {
		let url;
		if (queryParams) {
			// read in query parameters to set on the URL
			const params = new URLSearchParams();
			for (let key in queryParams) {
			  if (queryParams.hasOwnProperty(key)) {
			    const value = queryParams[key];
			    params.append(key, value);
			  }
			}
			url = API_URL + path + '?' + params.toString();
		} else {
			url = API_URL + path;
		}

		axios({
		  method: 'GET',
		  url: url,
		  headers: {
		    'Authorization': 'Bearer ' + token,
		  },
		}).then(response => {
		  console.log('response:', response.data)
		}).catch(error => {
		  console.log('error:', error.message)
		})
	},
	patch: function ({ path, token, body = {} }) {
		let url = API_URL + path;
		console.log(`PATCH request to url=${url} body=${body}`);
		axios({
		  method: 'PATCH',
		  url: url,
		  data: body,
		  headers: {
		    'Authorization': 'Bearer ' + token,
		  },
		}).then(response => {
		  console.log('response:', response.data)
		}).catch(error => {
		  console.log('error:', error.message, error.config.data)
		})
	},
	post: function ({ path, token, body = {} }) {
		let url = API_URL + path;
		console.log(`POST request to url=${url} body=${body}`);
		axios({
		  method: 'POST',
		  url: url,
		  data: body,
		  headers: {
		    'Authorization': 'Bearer ' + token,
		  },
		}).then(response => {
		  console.log('response:', response.data)
		}).catch(error => {
		  console.log('error:', error.message, error.config.data)
		})
	}
}
